core/templates/pages/exploration-player-page/services/exploration-recommendations.service.spec.ts

```diff
@@ -0,0 +1,206 @@
+  'pages/exploration-player-page/services/exploration-recommendations.service';
+import { ContextService } from 'services/context.service';
+import { ServicesConstants } from 'services/services.constants';
+import { UrlService } from 'services/contextual/url.service';
+import { ExplorationRecommendationsBackendApiService } from
+  'domain/recommendations/exploration-recommendations-backend-api.service';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { LearnerExplorationSummary } from 'domain/summary/learner-exploration-summary.model';
+
+
+
+describe('Exploration Recommendations Service', () => {
+  let expRecsService: ExplorationRecommendationsService;
+  let urlService: UrlService;
+  let contextService: ContextService;
+
+  const STORY_ID = '1';
+  const COLLECTION_ID = '2';
+  const NODE_ID = '3';
+  const EXPLORATION_ID = '4';
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+      ],
+      providers: [
+        ExplorationRecommendationsService,
+      ]
+    });
+  }));
+
+  beforeEach(() => {
+    urlService = TestBed.get(UrlService);
+    contextService = TestBed.get(ContextService);
+
+    spyOn(urlService, 'getCollectionIdFromExplorationUrl').and.returnValue(
+      COLLECTION_ID);
+
+    spyOn(urlService, 'getUrlParams').and.returnValue({
+      story_id: STORY_ID,
+      node_id: NODE_ID
+    });
+
+    spyOn(contextService, 'getExplorationId').and.returnValue(EXPLORATION_ID);
+  });
+
+  describe('when used in the editor', () => {
+    beforeEach(() => {
+      spyOn(contextService, 'getPageContext').and.returnValue(
+        ServicesConstants.PAGE_CONTEXT.EXPLORATION_EDITOR
+      );
+    });
+
+    it('should initialize with editor context', () => {
+      expRecsService = TestBed.get(ExplorationRecommendationsService);
+      expect(expRecsService.isInEditorPage).toBeTrue();
+    });
+
+    describe('in the Preview tab', () => {
+      beforeEach(() => {
+        spyOn(contextService, 'getEditorTabContext').and.returnValue(
+          ServicesConstants.EXPLORATION_EDITOR_TAB_CONTEXT.PREVIEW
+        );
+      });
+
+      it('should initialize with editor preview context', () => {
+        expRecsService = TestBed.get(ExplorationRecommendationsService);
+        expect(expRecsService.isInEditorPage).toBeTrue();
+        expect(expRecsService.isInEditorPreviewMode).toBeTrue();
+      });
+    });
+  });
+
+  describe('when used outside of the editor', () => {
+    beforeEach(() => {
+      spyOn(contextService, 'getPageContext').and.returnValue(
+        ServicesConstants.PAGE_CONTEXT.OTHER
+      );
+    });
+
+    it('should not initialize with editor context', () => {
+      expRecsService = TestBed.get(ExplorationRecommendationsService);
+      expect(expRecsService.isInEditorPage).toBeFalse();
+    });
+  });
+
+  let expRecsBackendApiService: ExplorationRecommendationsBackendApiService;
```

Comment:
All these should probably go in the beforeEach() below instead?

Also, I suggest having something like AUTHOR_REC_IDS = ['author_rec_5', 'author_rec_6'] or similar -- that way, easy to see what "5" and "6" are at a glance. Have it right next to authorRecommendations.

Reply:
did you mean they should go in the describe?
everything that's not constant is initialized in the beforeEach